src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -411,6 +411,19 @@ private static ImmutableArray<ISymbol> CreateInterfaceMembers(IEnumerable<ISymbo
                         break;
                     case SymbolKind.Property:
                         var property = member as IPropertySymbol;
+                        IMethodSymbol getMethod = null;
+                        var hasGetMethod = property.GetMethod != null && property.GetMethod.DeclaredAccessibility == Accessibility.Public;
+                        if (hasGetMethod)
+                        {
+                            // We recreate the get accessor because it is possible it has the readonly modifier due
+                            // to being an auto property on a struct which is invalid for an interface member
+                            getMethod = CodeGenerationSymbolFactory.CreateAccessorSymbol(
+                                property.GetMethod,
+                                property.GetMethod.GetAttributes(),
+                                property.GetMethod.DeclaredAccessibility,
+                                property.GetMethod.ExplicitInterfaceImplementations);
+                        }
```

Comment:
could you just call: CreateAccessorSymbol(property.GetMethod)?  it looks like that has the same result.  (you may need to get the attributes as well).  but the accessibility and explicit interface impls will be taken care of for you.

Reply:
done, kept the attributes.